python/pfs/utils/makePfsDesign.py

```diff
@@ -0,0 +1,110 @@
+                      fiberFlux=np.NaN, psfFlux=np.NaN, totalFlux=np.NaN,
+                      fiberFluxErr=np.NaN, psfFluxErr=np.NaN, totalFluxErr=np.NaN,
+                      filterNames=None, guideStars=None):
+    """ Make PfsDesign object from cobra x and y required positions.
+
+        Parameters
+        ----------
+        x : array_like
+            cobra x position (1..2394).
+        y: array_like
+            cobra y position (1..2394).
+
+        Returns
+        -------
+        pfsDesign : pfs.datamodel.PfsDesign
+           constructed pfsDesign for all fibers (science + engineering)
+    """
+
+    # Grand Fiber Map
+    gfm = FiberIds()
+    isEmpty = gfm.scienceFiberId == FiberIds.EMPTY
+    isEng = gfm.scienceFiberId == FiberIds.ENGINEERING
+    isCobra = ~isEmpty & ~isEng
+
+    nFiber = len(gfm.scienceFiberId)
+    nScienceFiber =  len(gfm.scienceFiberId[isCobra])
+    fiberId = gfm.fiberId
+
+    def setDefaultValues(sciVal, engVal, shape=nFiber, dtype=float):
+        """assign provided sci values to cobra index and default value to engineering fibers. """
+        array = np.empty(shape, dtype=dtype)
+
+        if not isinstance(sciVal, (float, str)):
+            sciVal = np.array(sciVal)
+            if sciVal.ndim == 1 and array.ndim == 2:
+                sciVal = sciVal[:, None]
+
+        array[isCobra] = sciVal
+        array[isEng] = engVal
+        # we also need to fill empty fibers with some values
+        array[isEmpty] = engVal
+
+        return array
+
+    tract = setDefaultValues(sciVal=tract, engVal=-1, dtype='int32')
+    patch = setDefaultValues(sciVal=patch, engVal='0,0', dtype='<U32')
+
+    if ra is None:
+        ra = raBoresight + 1e-3 * x
+
+    if dec is None:
+        dec = decBoresight + 1e-3 * y
+
+    ra = setDefaultValues(sciVal=ra, engVal=100)
+    dec = setDefaultValues(sciVal=dec, engVal=100)
+
+    catId = setDefaultValues(sciVal=catId, engVal=-1, dtype='int32')
+    objId = setDefaultValues(sciVal=objId, engVal=-1, dtype='int64')
+
+    targetType = setDefaultValues(sciVal=targetType, engVal=TargetType.ENGINEERING, dtype=int)
+    fiberStatus = setDefaultValues(sciVal=fiberStatus, engVal=FiberStatus.GOOD)
+
+    # I might be overaccommodating here but ...
+    if filterNames is None:
+        # making sure some input data are not discarded silently
+        assert fiberFlux==np.NaN and psfFlux==np.NaN and totalFlux==np.NaN
+        filterNameArray = np.array(nScienceFiber * [[]], dtype='str')  # list of filter names
+
+    elif isinstance(filterNames, str):
+        filterNameArray = np.array(nScienceFiber * [[filterNames]])
+
+    else:
+        filterNameArray = np.array(filterNames)
+        filterNameArray = filterNameArray[:, None] if filterNameArray.ndim == 1 else filterNameArray
+
+    nFilter = filterNameArray.shape[1]
+    filterNameArray = setDefaultValues(sciVal=filterNameArray, engVal='', dtype=filterNameArray.dtype,
```

Comment:
I don't think `''` is the right value for engineering fibers. I think nothing is better. I.e., engineering fibers shouldn't have any `filterNames` set. This is why `PfsDesign` treats `filterNames` as a list and not a numpy array.